### DISCUSSION TITLE
dnscrypt-proxy unable to resolve onion address

### DISCUSSION THREAD

**Discussion by xxjoe2** [DNSCrypt/dnscrypt-proxy#2356] [Q&A] [answered]
i want to use dnscrypt-proxy to handle all dns queries, it works except resolving onion address. i have tried some previous versions as well as the latest version

i have set in toml:
force_tcp to true
proxy = 'socks5://127.0.0.1:9050'
sources.'onion-services' added

but when i use try:
dnscrypt-proxy -config /tmp/dnscrypt-proxy2/dnscrypt-proxy-tor.toml -resolve www.facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion

it returns "Name servers  : name does not exist"

what could be the problem? i have been searching for a few days but there's not too much about this

thanks

**Comment by jedisct1** [comment]
Onion addresses don't use DNS, they don't have IP addresses !

**Reply by xxjoe2** [reply]
sorry, i have modified my question above, it was based on my previous tests as below:

dig +short cnn.com @169.254.0.1 -p 9053 (9053 is Tor's DNSPort)
151.101.67.5

dig +short www.facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion @169.254.0.1 -p 9053
172.29.16.186

dig +short duckduckgogg42xjoc72x3sjasowoarfbgcmvfimaftt6twagswwzczad.onion @169.254.0.1 -p 9053
172.19.168.148


so what should i do in order to let dnscrypt proxy to resolve onion addresses?

**Comment by jedisct1** [comment, marked as answer]
In your `forwarding-rules.txt` file, put

```
onion             169.254.0.1:9053
```

**Reply by xxjoe2** [reply]
oh, it works now

hardly find any docs mentioned about this

thanks a lot
